core/dbt/parser/schemas.py

```diff
@@ -110,29 +121,37 @@ def _trimmed(inp: str) -> str:
     return inp[:44] + '...' + inp[-3:]
 
 
+def merge_freshness(
+    base: Optional[FreshnessThreshold], update: Optional[FreshnessThreshold]
+) -> Optional[FreshnessThreshold]:
+    if base is not None and update is not None:
```

Comment:
Actually, and this gets me almost every time I go to edit this stuff, if `base is not None` and `update is None`, then we want to get `None`, because it means the user explicitly did something like:
```
source:
 - name: my_source
    freshness:
      warn_after: {count: 10, period: hour}
      error_after: {count: 1, period: day}
    tables:
      - name: my_table
         # disable freshness checks for this table
         freshness: null
```

If they had just not specified freshness, they'd get a non-None `FreshnessThreshold` where all the members are `None`.